docs/FACTORY_RESET.md

```diff
@@ -0,0 +1,136 @@
+  kernel command line option (`systemd.factory_reset=1`) or via the UEFI
+  variable `FactoryResetRequest` (see above). The
+  `systemd-factory-reset-generator` unit generator checks both these conditions
+  and adds `factory-reset-now.target` to the boot transaction, already in the
+  initial RAM disk (initrd).
+
+* The
+  [`systemd-factory-reset-complete.service`](https://www.freedesktop.org/software/systemd/man/latest/systemd-factory-reset-complete.service.html)
+  unit is invoked after `factory-reset-now.target` and marks the factory reset
+  operation as complete. The boot process then may continue.
+
+* The
+  [`systemd-repart`](https://www.freedesktop.org/software/systemd/man/latest/systemd-repart.html)
+  tool can take the factory reset logic into account. Either on explicit
+  request via the `--factory-reset=` logic, or automatically derived from the
+  aforementioned kernel command line switch and EFI variable. When invoked for
+  factory reset it will securely erase all partitions marked for that via the
+  `FactoryReset=` setting in its partition definition files. Once that is
+  complete it will execute the usual setup operation, i.e. format new
+  partitions again.
+
+* The
+  [`systemd-logind.service(8)`](https://www.freedesktop.org/software/systemd/man/latest/systemd-logind.service.html)
+  unit supports automatically binding factory reset to special keypresses
+  (typically long presses), see the
+  [`logind.conf(5)`](https://www.freedesktop.org/software/systemd/man/latest/logind.conf.html)
+  man page.
+
+* The
+  [`systemd-factory-reset`](https://www.freedesktop.org/software/systemd/man/latest/systemd-factory-reset.html)
+  tool can be used to query the current state of the factory request mechanism,
+  i.e. whether a factory reset is currently being executed, or one has been
+  requested for the next boot.
+
+* The `/run/systemd/io.systemd.FactoryReset` Varlink service provides two IPC
+  APIs for working with factory reset: it permits querying whether the local
+  system supports requesting a factory reset by starting
+  `factory-reset.target`. This may be used by UIs to hide or show in the UI an
+  interface to request a factory reset. The Varlink IPC service also reports
+  the current factory reset state, much like the `systemd-factory-reset` tool
+  mentioned above. This may be used by various early boot services that
+  potentially intent to reset system state during a factory reset operation.
+
+## Exposure in the UI
+
+If a graphical UI shall expose a factory reset operation it should first check
+if requesting a factory reset is supported at all via the Varlink service
+mentioned above. Once a factory reset shall be executed it shall ask for
+activation of the `factory-reset.target` unit.
+
+Alternatively, `systemd-logind.service`'s hotkey support may be used, for
+example to request factory reset if the power button is pressed for a long
+time.
```

Comment:
Giving the power button as an example seems like a really bad idea. Please make it any other button that users aren't likely to press for a long time.